### PR TITLE
Update dependencies & setup life-cycle

### PR DESCRIPTION
**Pull request by shokmaster**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,34 +1,77 @@
...
   directories:
     - $HOME/.npm
-    - $HOME/.cache  # includes bowers cache
-    - node_modules
 
-matrix:
-  fast_finish: true
+env:
+  global:
+    - JOBS=1
```
Comment (josex2r) [inline .travis.yml:16]: Add the env var `CI=true`, it disables the disk cache on CI environments.
[Link to CI](https://github.com/stefanpenner/broccoli-persistent-filter#persistent-cache)

**Review by josex2r — DISMISSED** [review]

File: .travis.yml (hunk 1 above)
Comment (adrigzr) [reply .travis.yml:16]: There is no need for that. Travis has it has default environment variable:

https://docs.travis-ci.com/user/environment-variables/#default-environment-variables

File: .travis.yml [hunk 2]
```diff
@@ -1,34 +1,77 @@
...
+        - commitlint-travis
+        - npm run lint:js
+        - npm run lint:hbs
+        - npm test
+      after_script:
+        - greenkeeper-lockfile-upload
+      after_success:
+        - codecov --file=coverage/lcov.info
+
+    #- stage: "Regression Tests"
```
Comment (adrigzr) [inline .travis.yml]: Given that currently tests are only node tests, you need to add a node version matrix:

https://github.com/BBVAEngineering/broccoli-image-meta/blob/b803400c5e381ca8e78abfa3ef7970c71be85346/.travis.yml#L47

File: .travis.yml [hunk 3]
```diff
@@ -1,34 +1,77 @@
...
-  - npm --version
+#before_install:
+#  - npm config set spin false
+#  - npm install -g npm@6
+#  - npm --version
 
-install:
-  - npm ci
+#script:
+#  - node_modules/.bin/ember try:one $EMBER_TRY_SCENARIO
```
Comment (adrigzr) [inline .travis.yml]: Change this for `npm test` with node version matrix.

File: lib/.eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+'use strict';
+
+module.exports = {
+	env: {
+		node: true
```
Comment (adrigzr) [inline lib/.eslintrc.js]: This can be added to root `.eslintrc`.

File: package.json [hunk 5]
```diff
@@ -10,11 +10,19 @@
...
-    "test": "npm run mocha",
-    "lint": "node_modules/.bin/eslint '**/*.js'",
+    "lint:hbs": "ember-template-lint .",
+    "lint:js": "eslint --max-warnings 0 ./*.js lib tests",
     "check": "node_modules/.bin/check-dependencies",
-    "mocha": "node_modules/.bin/mocha --recursive tests"
+    "pretest": "npm run check && npm run lint:js",
+    "test": "npm run clean-coverage && npm run test:node",
+    "test:node": "nyc --reporter=html --reporter=json --reporter=json-summary --reporter=lcov --report-dir coverage --exclude tests mocha tests",
+    "mocha": "node_modules/.bin/mocha --recursive tests",
```
Comment (adrigzr) [inline package.json:19]: This task is never called.

File: package.json [hunk 6]
```diff
@@ -10,11 +10,19 @@
...
-    "pretest": "npm run check && npm run lint",
-    "test": "npm run mocha",
-    "lint": "node_modules/.bin/eslint '**/*.js'",
+    "lint:hbs": "ember-template-lint .",
+    "lint:js": "eslint --max-warnings 0 ./*.js lib tests",
     "check": "node_modules/.bin/check-dependencies",
-    "mocha": "node_modules/.bin/mocha --recursive tests"
+    "pretest": "npm run check && npm run lint:js",
+    "test": "npm run clean-coverage && npm run test:node",
+    "test:node": "nyc --reporter=html --reporter=json --reporter=json-summary --reporter=lcov --report-dir coverage --exclude tests mocha tests",
```
Comment (adrigzr) [inline package.json]: You missed `--recursive` option (dont know if needed).

File: package.json [hunk 7]
```diff
@@ -10,11 +10,19 @@
...
   "scripts": {
-    "pretest": "npm run check && npm run lint",
-    "test": "npm run mocha",
-    "lint": "node_modules/.bin/eslint '**/*.js'",
+    "lint:hbs": "ember-template-lint .",
+    "lint:js": "eslint --max-warnings 0 ./*.js lib tests",
     "check": "node_modules/.bin/check-dependencies",
-    "mocha": "node_modules/.bin/mocha --recursive tests"
+    "pretest": "npm run check && npm run lint:js",
+    "test": "npm run clean-coverage && npm run test:node",
```
Comment (adrigzr) [inline package.json]: `clean-coverage` should be added to `pretest`.

File: tests/.eslintrc.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+'use strict';
+
+module.exports = {
+	env: {
+		mocha: true
```
Comment (adrigzr) [inline tests/.eslintrc.js:5]: Mocha tests should be moved to `node-tests` folder as we do in other addons.

This file can be moved to root `.eslintrc`.

**Review by adrigzr — CHANGES_REQUESTED** [review]

File: lib/.eslintrc.js (hunk 4 above)
Comment (next-juanantoniogomez) [reply lib/.eslintrc.js]: done

File: package.json (hunk 5 above)
Comment (next-juanantoniogomez) [reply package.json:19]: included in "test:node"

File: tests/.eslintrc.js (hunk 8 above)
Comment (next-juanantoniogomez) [reply tests/.eslintrc.js:5]: Yeah, we plan to rename this folder when we include ember tests

File: .travis.yml (hunk 3 above)
Comment (next-juanantoniogomez) [reply .travis.yml]: done

File: .travis.yml (hunk 2 above)
Comment (next-juanantoniogomez) [reply .travis.yml]: done

File: package.json (hunk 6 above)
Comment (next-juanantoniogomez) [reply package.json]: fixed calling to "test:node"

File: package.json (hunk 7 above)
Comment (next-juanantoniogomez) [reply package.json]: moved!

**Review by adrigzr — APPROVED** [review]

**Pull request merged**